packages/nextjs-kit/src/components/footer.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Link from 'next/link';
+import { FooterMenuItem } from '../types/index';
+
+interface FooterMenuProps {
+  footerMenuItems: FooterMenuItem[];
+  children: JSX.Element;
+}
+
+/**
+ * This is a Footer component.
+ *
+ * @param {Props} footerMenuItems holds the linkText, href, and parent of each route to be displayed
+ * @param {Props} children  copyright display and information to be passed in by the user
+ * @returns {React.JSX.Element}
+ */
+const Footer: React.FC<FooterMenuProps> = ({
+  footerMenuItems,
+  children,
+}: FooterMenuProps) => {
+  const FooterMenu = () => {
+    const menuArr = [];
+    if (footerMenuItems) {
+      // some not so great code to account for nested menu elements
+      for (let i = 0; i < footerMenuItems.length; i++) {
+        if (footerMenuItems[i + 1] && footerMenuItems[i + 1].parent) {
```

Comment:
this wouldn't work for the WordPress data that includes `parentId`. This might be a good opportunity to try a type predicate:

```ts
const hasParent = (item: FooterMenuItem): item is FooterMenuItem =>
  item.parentId ? true : item.parent ? true : false;
```

Then, in the loop we can check if `parent` exists:

```ts
const itemHasParent = hasParent(footerMenuItems[i + 1]);
```

Reply:
Was spending too much time on the CR revisions for the CSS on the child component so pushed without a solution. Can maybe circle back on this next week?